sdk/webpubsub/azure-webpubsub-client/azure/webpubsub/client/_models.py

```diff
@@ -0,0 +1,569 @@
+        if isinstance(data, (bytes, bytearray)):
+            return base64.b64encode(data).decode()
+        raise TypeError("Message must be a bytes or bytearray")
+    raise TypeError(f"Unsupported dataType: {data_type}")
+
+
+def parse_payload(data: Any, data_type: WebPubSubDataType) -> Any:
+    if data_type == WebPubSubDataType.TEXT:
+        if not isinstance(data, str):
+            raise TypeError("Message must be a string when dataType is text")
+        return data
+    if data_type == "json":
+        return data
+    if data_type in ("binary", "protobuf"):
+        if isinstance(data, (bytes, bytearray)):
+            return data
+        return bytes(base64.b64decode(data))
+    raise TypeError(f"Unsupported dataType: {data_type}")
+
+
+class WebPubSubClientProtocol:
+    def __init__(self) -> None:
+        self.is_reliable_sub_protocol = False
+        self.name = ""
+
+    @staticmethod
+    def parse_messages(
+        raw_message: str,
+    ) -> Union[ConnectedMessage, DisconnectedMessage, GroupDataMessage, ServerDataMessage, AckMessage]:
+        if raw_message is None:
+            raise Exception("No input")
+        if not isinstance(raw_message, str):
+            raise Exception("Invalid input for JSON hub protocol. Expected a string.")
+
+        message = json.loads(raw_message)
+        if message["type"] == "system":
+            if message["event"] == "connected":
+                return ConnectedMessage(
+                    connection_id=message["connectionId"],
+                    user_id=message["userId"],
+                    reconnection_token=message.get("reconnectionToken"),
+                )
+            if message["event"] == "disconnected":
+                return DisconnectedMessage(message=message["message"])
+            raise Exception(f"wrong message event type: {message['event']}")
+        if message["type"] == "message":
+            if message["from"] == "group":
+                data = parse_payload(message["data"], message["dataType"])
+                return GroupDataMessage(
+                    data_type=message["dataType"],
+                    data=data,
+                    group=message["group"],
+                    from_user_id=message["fromUserId"],
+                    sequence_id=message.get("sequenceId"),
+                )
+            if message["from"] == "server":
+                data = parse_payload(message["data"], message["dataType"])
+                return ServerDataMessage(
+                    data=data, data_type=message["dataType"], sequence_id=message.get("sequenceId")
+                )
+            raise Exception(f"wrong message from: {message['from']}")
+        if message["type"] == "ack":
+            return AckMessage(
+                ack_id=message["ackId"],
+                success=message["success"],
+                error=message.get("error"),
+            )
+        raise Exception(f"wrong message type: {message['type']}")
```

Comment:
Consider returning null and log instead of throw Exception for unknown type or event for forward compatibility. Only throw when parsing has error